compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1000,6 +1000,8 @@ class Namer { typer: Typer =>
           ) match
               case Some(other) => No(i"overrides ${other.showLocated}, which is already a member of $cls")
               case None => Yes
+        else if sym.is(JavaDefined) then
```

Comment:
I think the issue comes from trying to export a Java class because we're missing code for exporting the static methods of the class but we shouldn't prevent exporting a Java method